aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -660,8 +660,7 @@ Tensor& logsumexp_out(Tensor& result, const Tensor& self, DimnameList dims, bool
 
 static Tensor& norm_out(Tensor &result, const Tensor &self, optional<Scalar> opt_p,
                                IntArrayRef dim, bool keepdim, optional<ScalarType> opt_dtype) {
-  auto p = opt_p.value_or(2.0);
-  TORCH_CHECK(!(p.toDouble() == 2 && self.is_complex()), "norm with p=2 not supported for complex tensors");
+  auto p = opt_p.value_or(2.0).to<double>();
```

Comment:
What happens if opt_p is complex?

I imagine a lot of our code is casting to double naively.

Reply:
Looks like attempting to cast complex to double throws an error:
```
>>> torch.linalg.norm(torch.randn(10), ord=1j)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: value cannot be converted to type double without overflow: (0,1)
```